### PR TITLE
[Task-2] Implement Top CPU process log; add log to file

### PR DESCRIPTION
**Pull request by Shagon1k**

### CODE REVIEW

File: task-2/package.json [hunk 1]
```diff
@@ -0,0 +1,21 @@
+{
+    "name": "task-2",
+    "version": "1.0.0",
+    "description": "",
+    "main": "index.js",
+    "dependencies": {
+        "undici-types": "^5.26.5"
```
Comment (Anton-Koniouchevsky) [inline task-2/package.json]: is this used anywhere?

File: task-2/src/index.ts [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+        console.error("Error during command execution.", stderr);
+      }
+
+      if (error !== null) {
+        console.error("Unsuccessful run command.", error);
+      }
+
+      // Re-run based on min refresh rate: either rate achieved OR immediately if command execution lasts too long
+      const executionTime = executionEndTime - executionStartTime;
+      if (executionTime > minCommandIntervalMs) {
```
Comment (Anton-Koniouchevsky) [inline task-2/src/index.ts]: It seems like we can avoid setting a zero timeout when executionTime === minCommandIntervalMs
```suggestion
      if (executionTime >= minCommandIntervalMs) {
```

File: task-2/src/index.ts [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+          lastLogFileTime = Date.now();
+        }
+      }
+
+      if (stderr) {
+        console.error("Error during command execution.", stderr);
+      }
+
+      if (error !== null) {
+        console.error("Unsuccessful run command.", error);
```
Comment (Anton-Koniouchevsky) [inline task-2/src/index.ts]: note: Not sure if it's necessary or even possible to continue running the script if an error occurs.

File: task-2/src/index.ts [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+  console.log(result);
+};
+
+const runCommand = (
+  command: string,
+  minRefreshRate: number,
+  logFileIntervalMs: number
+) => {
+  const minCommandIntervalMs = 1000 / minRefreshRate;
+  let lastLogFileTime = Date.now();
```
Comment (Anton-Koniouchevsky) [inline task-2/src/index.ts]: This Date.now() leads to the fact that we do not write the first result to the file after calling the script. This is not explicitly described in the requirements, but writing down the first result seems logical to me.

Comment (Shagon1k) [reply task-2/src/index.ts]: Seems logical for me either, thought about it during implementation either.
Fixed.

File: task-2/src/index.ts (hunk 3 above)
Comment (Shagon1k) [reply task-2/src/index.ts]: Throwing error or return from function seems like logical handle here, right?
Fixed

File: task-2/src/index.ts (hunk 2 above)
Comment (Shagon1k) [reply task-2/src/index.ts]: Fixed.

File: task-2/package.json (hunk 1 above)
Comment (Shagon1k) [reply task-2/package.json]: To be honest, cant understand how this was added (I even don't know what is it :D). Probably some side-effect of adding "@types/node", which is strange but the only guess.
Removed.

**Review by Anton-Koniouchevsky — APPROVED** [review]

**Pull request merged**